src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.cs

```diff
@@ -161,6 +161,31 @@ public async Task LengthIsNotCachedAfterHandleHasBeenExposed(FileAccess fileAcce
             Assert.Equal(2, stream.Length);
             Assert.Equal(2, createdFromHandle.Length);
         }
+
+        [Fact]
+        public async Task WriteByteFlushesTheBufferWhenItBecomesFull()
+        {
+            string filePath;
+            List<byte> writtenBytes = new List<byte>();
+
+            using (FileStream stream = (FileStream)await CreateWriteOnlyStreamCore(Array.Empty<byte>()))
+            {
+                filePath = stream.Name;
+
+                stream.WriteByte(0);
+                writtenBytes.Add(0);
+
+                byte[] bytes = new byte[BufferSize - 1];
```

Comment:
Is it worth putting something other than zeros in here? So it's distinct from the first byte you wrote

Reply:
Doesn't really matter. What matters is that we are attempting to write an 11th character on the stream when the buffer can only hold 10 characters.

Reply:
...and the problem is that the internal buffer is not being reset/flushed when WriteByte is called when this buffer is already full.